src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -332,42 +345,21 @@ class ShenandoahHeap : public CollectedHeap {
-  // two values are constant throughout each GC phases, we introduce a new service into ShenandoahGeneration.  This service
-  // provides adjusted_available() based on an adjusted capacity.  At the start of evacuation, we adjust young capacity by
-  // adding the amount to be borrowed from old-gen and subtracting the _young_evac_reserve, we adjust old capacity by
-  // subtracting the amount to be loaned to young-gen.
-  //
-  // We always use adjusted capacities to determine permission to allocate within young and to promote into old.  Note
-  // that adjusted capacities equal traditional capacities except during evacuation and update refs.
-  //
-  // During evacuation, we assure that _old_evac_expended does not exceed _old_evac_reserve.
-  //
-  // At the end of update references, we perform the following bookkeeping activities:
-  //
-  // 1. Unadjust the capacity within young-gen and old-gen to undo the effects of borrowing memory from old-gen.  Note that
-  //    the entirety of the collection set is now available, so allocation capacity naturally increase at this time.
-  // 2. Clear (reset to zero) _alloc_supplement_reserve, _young_evac_reserve, _old_evac_reserve, and _promoted_reserve
-  //
-  // _young_evac_reserve and _old_evac_reserve are only non-zero during evacuation and update-references.
-  //
-  // Allocation of old GCLABs assures that _old_evac_expended + request-size < _old_evac_reserved.  If the allocation
-  //  is authorized, increment _old_evac_expended by request size.  This allocation ignores old_gen->available().
+  // TODO: Revisit the following comment.  It may not accurately represent the true behavior when evacuations fail due to
+  // difficulty finding memory to hold evacuated objects.
   //
   // Note that the typical total expenditure on evacuation is less than the associated evacuation reserve because we generally
   // reserve ShenandoahEvacWaste (> 1.0) times the anticipated evacuation need.  In the case that there is an excessive amount
   // of waste, it may be that one thread fails to grab a new GCLAB, this does not necessarily doom the associated evacuation
   // effort.  If this happens, the requesting thread blocks until some other thread manages to evacuate the offending object.
   // Only after "all" threads fail to evacuate an object do we consider the evacuation effort to have failed.
 
-  intptr_t _alloc_supplement_reserve;  // Bytes reserved for young allocations during evac and update refs
   size_t _promoted_reserve;            // Bytes reserved within old-gen to hold the results of promotion
   volatile size_t _promoted_expended;  // Bytes of old-gen memory expended on promotions
 
+  // Allocation of old GCLABs (aka PLABs) assures that _old_evac_expended + request-size < _old_evac_reserved.  If the allocation
+  //  is authorized, increment _old_evac_expended by request size.  This allocation ignores old_gen->available().
+
```

Comment:
This documentation comment seems like a stream-of-consciousness thing. I think we need this comment to move to the right spot.